tests/SlackConnector.Tests.Unit/Connections/Clients/Flurl/FlurlChannelClientTests.cs

```
@@ -64,6 +65,166 @@ public void Dispose()
+            _httpTest
+                .ShouldHaveCalled(ClientConstants.SlackApiHost.AppendPathSegment(FlurlChannelClient.CHANNEL_CREATE_PATH))
+                .WithQueryParamValue("token", slackKey)
+                .WithQueryParamValue("name", channelName)
+                .Times(1);
+
+            result.ToExpectedObject().ShouldEqual(expectedResponse.Channel);
+        }
+
+        [Fact]
+        public async Task should_join_channel()
+        {
+            // given
+            const string slackKey = "I-is-another-key";
+            const string channelName = "my-channel";
+
+            var expectedResponse = new ChannelResponse()
+            {
+                Channel = new Channel
+                {
+                    Id = "some-channel",
+                    IsChannel = true
+                }
+            };
+
+            _httpTest.RespondWithJson(expectedResponse);
+
+            // when
+            var result = await _channelClient.JoinChannel(slackKey, channelName);
+
+            // then
+            _responseVerifierMock.Verify(x => x.VerifyResponse(Looks.Like(expectedResponse)), Times.Once);
+            _httpTest
+                .ShouldHaveCalled(ClientConstants.SlackApiHost.AppendPathSegment(FlurlChannelClient.CHANNEL_JOIN_PATH))
+                .WithQueryParamValue("token", slackKey)
+                .WithQueryParamValue("name", channelName)
+                .Times(1);
+
+            result.ToExpectedObject().ShouldEqual(expectedResponse.Channel);
+        }
+
+        [Fact]
+        public async Task should_archive_channel()
+        {
+            // given
+            const string slackKey = "I-is-another-key";
+            const string channelName = "my-channel";
+
+            var expectedResponse = new StandardResponse()
+            {
+                Ok = true
+            };
+
+            _httpTest.RespondWithJson(expectedResponse);
+
+            // when
+            var result = true;
+            try
```

Should let exceptions bubble up and be caught by the test harness